File: frontend/js/chat/models/DisconnectMessage.js
export default class DisconnectMessage {

    constructor(username, type="disconnect_message"){
        this.username = username;
        this.type = type;
    }

    static fromData(data){
        return new DisconnectMessage(
            data.username,
            data.type
        )
    }

    toData(){
        return {
            username: this.username,
            type: this.type
        }
    }

    getUsername(){
        return this.username;
    }

    getType(){
        return this.type;
    }
}

File: backend/models/disconnect_message.py
class DisconnectMessage:

    def __init__(self, username: str, type: str = "disconnect_message"):
        self._username = username
        self._type = type

    def to_data(self) -> dict:
        return {
            "username": self.get_username(),
            "type": self.get_type()
        }

    def get_username(self) -> str:
        return self._username

    def get_type(self) -> str:
        return self._type